### PR TITLE
Add initial version of workflow

### PR DESCRIPTION
**Pull request by uturuncoglu**
This PR aims to add initial version of end-to-end DATM+SCHISM workflow using UWTools.

### CODE REVIEW

**Comment by uturuncoglu** [comment]
@janahaddad @saeed-moghimi-noaa @mansurjisan @yunfangsun @danishyo JFYI, this PR is ready to merge and we could start testing workflow, make adjustments and fix the bugs based on user feedbacks. It will also update the  documentation and include new section about usage of the workflow. So, I'll do last round of testing before merge tomorrow.

**Comment by uturuncoglu** [comment]
@janahaddad @saeed-moghimi-noaa @mansurjisan @yunfangsun @danishyo This is also tested on Hercules only. I could try on Frontera too.

**Comment by uturuncoglu** [comment]
BTW, we have still issue with SCHSIM - https://github.com/oceanmodeling/ufs-weather-model/issues/146

File: ush/coastal.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          subset: true
+          combine: false
+          target_directory: 'INPUT'
+  template_file: ../templates/cdeps.streams
+schism:
+  hgrid: '{{ dir.data }}/hgrid.gr3'
+  vgrid: '{{ dir.data }}/vgrid.in'
+  boundary:
+    vars: [True, True, True]
+    ids: [0]
```
Comment (mansurjisan) [inline ush/coastal.yaml:144]: @uturuncoglu ,

the boundary generation should be done through gen_bctides.py. The gen_bnd.py file is primarily used to create boundary conditions from HYCOM data. In the current version of gen_bctides.py we have the capability to define HYCOM as boundary data source. So, I don't think we need to include gen_bnd.py in workflow. Also, in the regression test we don't use HYCOM as boundary data source; we are using tidal boundary conditions.

File: ush/utils/schism/gen_bnd.py
Comment (mansurjisan) [inline ush/utils/schism/gen_bnd.py:1]: We no longer need this script since **gen_bctides.py** has the HYCOM data download capability built into the script.

File: ush/coastal.yaml (hunk 1 above)
Comment (uturuncoglu) [reply ush/coastal.yaml:136]: @mansurjisan Thanks for help. At this point, I configured `boundary` and `bctides` sections as optional. So, user does not need to provide those options and skip them. The workflow will just generate `*.gr3` files. We could revisit this part after team start using it.

**Comment by uturuncoglu** [comment]
@mansurjisan @janahaddad Thanks for your help. I have just merged this PR. I'll update the documentation we page soon. Then, we could start testing in UFS Coastal team.
